build.sbt

```diff
@@ -306,7 +327,7 @@ lazy val dagShared = (project in file("modules/dag-shared"))
 lazy val dagL1 = (project in file("modules/dag-l1"))
   .enablePlugins(AshScriptPlugin)
   .enablePlugins(JavaAppPackaging)
-  .dependsOn(kernel, shared % "compile->compile;test->test", dagShared, sdk, testShared % Test)
+  .dependsOn(dagShared % "compile->compile;test->test", sdk, kernel, testShared % Test)
```

Comment:
What about transitive dependencies? For example I'm almost sure that dagL1 depends on schema as well.

Reply:
in the current form, you were not able to handle all the dependencies and we used partially explicitly declared dependencies and partially implicitly (transitive). it's better to stay with one solution, so it would be way cleaner, to specify minimum set of libraries explicitly in the base package, and whatever is needed more on higher packages, should be declared separately. we cannot avoid transitive dependencies anyway, so let's use it instead of redeclaring them (and then noone is keeping an order there)

Reply:
I just remember that there was a discussion and we wanted to avoid transitive dependencies hence the comment. Given that external libraries pull in their versions of dependencies with them there is a chance that by removing some explicitly declared library from the base module, the version of this library in the depending module may change (because it's neither declared explicitly in the base module anymore but still can be available through base module dependencies and was never declared explicitly in the depending module). If we drop by a major version I guess it can mean trouble. Also in such a situation we are losing the compiler warning about possible incompatibilities afaik. Anyway I'm not sure what's the best approach here, just recalling the discussion we had some time ago and giving my 2 cents.

Reply:
We can run `sbt dependencyBrowseGraph` or `sbt dependencyTree` to find transitive dependencies eventually.

Reply:
@TheMMaciek I didn't remove dependencies from base modules (shared, schema, etc). I just removed redeclaration (probably not all, I'd keep it as a continuation process), so:
- if shared requires kryo, then schema don't need to redeclare kryo and can use it, because depends on shared.
- if kernel requires droste, but not shared, then kernel should declare droste dependency
- if shared depend on external library (let's say http4s) that is dependant on fs2, we should anyway declare fs2 explicitly.

so tl;dr - no transitive dependencies from external libraries. only between our internal modules